src/web/rustdoc.rs

```diff
@@ -223,6 +222,13 @@ pub fn rustdoc_html_server_handler(req: &mut Request) -> IronResult<Response> {
         req_path.remove(0);
     }
 
+    let redirect = |name: &str, vers: &str, path: &str| -> IronResult<Response> {
```

Comment:
nit: I think this would be better if it took a slice and called `.join()` here. It's not a blocker though.
```suggestion
    let redirect = |name: &str, vers: &str, path: &[str]| -> IronResult<Response> {
```